File: com.dexaint.dukechess/src/main/java/com/dexaint/dukechess/movement/Strike.java
package com.dexaint.dukechess.movement;

import org.apache.commons.lang3.ArrayUtils;
import com.dexaint.dukechess.flow.Field;
import com.dexaint.dukechess.flow.Player;

public class Strike extends MovementImpl {
	public int[] validateMovement(Field field, int pos, Destination d, Player p) {
		int[] ret = new int[]{};
		
		int[] offsets = d.dest2Offset(p);
		
		if (!isInField(field, pos, offsets) || hasMyChess(field, pos, offsets, p)) return ret; // Short-Circuit-OR
		
		ret = ArrayUtils.add(ret, offset2Dest(field, pos, offsets));
		
		return ret;
	}
}

File: deprecated/Unity/Assets/Scripts/Movement/Strike.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Strike : Movement
{
    public override List<int> ValidateMovement(List<ChessController> board, int pos, DestinationParser.Destination d, Player p)
    {
        List<int> ret = new List<int>();

        int[] offsets = DestinationParser.Dest2Offset(d, p);

        if (!IsInField(pos, offsets) || HasMyChess(board, pos, offsets, p) || !HasAnyChess(board, pos, offsets)) return ret; // Short-Circuit-OR

        ret.Add(Offset2Dest(pos, offsets));

        return ret;
    }
}
